[EDU-3247] feat: origin CRUD cli (#436)
aziontech/docs: src/content/docs/en/pages/devtools/cli/azion-cli-alpha/origins/origins.mdx

---
title: Origins

> description: Create, list, describe, update, and delete your origins on the Azion platform through Azion CLI.

Reviewer: ```suggestion
description: Create, list, describe, update, and delete your origins on Azion's platform through Azion CLI.
```

#### name

The `--name` flag sets the name of the origin being created.

#### addresses

The `--addresses` flag passes a list of addresses linked to the origin. Multiple addresses should be separated by commas.

### Optional flags

#### file

The `--file` flag specifies the path to a JSON file containing the attributes of the origin that will be created. You can use `-` for reading from stdin.

#### hmac-access-key

The `--hmac-access-key` flag informs the Hmac Access Key.

#### hmac-authentication

The `--hmac-authentication` flag specifies whether Hmac Authentication is used or not.

#### hmac-region-name

The `--hmac-region-name` flag informs the Hmac region name.

#### hmac-secret-key

The `--hmac-secret-key` flag informs the Hmac Secret Key.

#### host-header

The `--host-header` flag specifies the hostname of the server being accessed.

#### origin-path

The `--origin-path` flag defines the path to be appended to the URI when forwarding the request to the origin. Leave it blank to use only the URI.

#### origin-protocol-policy

The `--origin-protocol-policy` flag tells the protocol policy used in the origin.

#### origin-type

> The `--origin-type` flag identifies the source of a record, i.e., `single_origin`.

Reviewer: ```suggestion
The `--origin-type` flag identifies the source of a record. Example: `single_origin`.
```

Reviewer 2: isso, não usamos i.e.

### help

The `-h` or `--help` option displays more information about the `azion create origin` action.

---

## List

### Usage

```bash
$ azion list origin --application-id <application-id>
```

### Required flags

#### application-id

> The `--application-id` flag sets the unique identifier for the edge application. It is a required field.

Reviewer: ```suggestion
The `--application-id` flag sets the unique identifier for the edge application. It's a required field.
```

### Optional flags

#### details

The `--details` flag displays all relevant fields when listing origin.

#### filter

The `--filter` flag filters items by their name.

#### order-by

The `--order-by` flag sorts the output based on the selected field.

#### page

> The `--page` flag returns a specific page of the list according to its number. The default is 1.

Reviewer: ```suggestion
The `--page` flag returns a specific page of the list according to its number. The default is `1`.
```

#### page-size

> The `--page-size` flag defines how many items should be returned per page. The default is 10.

Reviewer: ```suggestion
The `--page-size` flag defines how many items should be returned per page. The default is `10`.
```

#### sort

> The `--sort` flag defines the order of the items on the list; options are `<asc|desc>`.

Reviewer: ```suggestion
The `--sort` flag defines the order of the items on the list. The value should be either `asc` or `desc`.
```

Reviewer: following the example in cache settings comm ref doc.

### help

The `-h` or `--help` option displays more information about the `azion list origin` action.

---

## Describe

### Usage

```bash
$ azion describe origin --application-id <application-id> --origin-key <origin-key> 
```

### Required flags

#### application-id

> The `--application-id` flag sets the unique identifier for an edge application. It is a mandatory field.

Reviewer: ```suggestion
The `--application-id` flag sets the unique identifier for an edge application. It's a mandatory field.
```

#### origin-key

> The `--origin-key` flag sets the unique identifier for an origin. It is a mandatory field.

Reviewer: ```suggestion
The `--origin-key` flag sets the unique identifier for an origin. It's a mandatory field.
```

### Optional flags

#### format

The `--format` flag changes the output format. Pass the value `json` to the flag to get the output in JSON format.

#### out

The `--out` flag exports the output to the given file path or filename with extension.

### help

The `-h` or `--help` option displays more information about the `azion describe origin` action.

---

## Update

### Usage

```bash
$ azion update origin --application-id <application-id> --origin-key <origin-key> 
```

### Required flags

#### application-id

> The `--application-id` flag sets the unique identifier for an edge application. It is a required field.

Reviewer 2: ```suggestion
The `--application-id` flag sets the unique identifier for an edge application. It 's a required field.
```

> The `--application-id` flag sets the unique identifier for an edge application. It is a required field.
> 
> #### origin-key
> 
> The `--origin-key` flag sets the unique identifier for an origin. It is a required field.
> 

Reviewer: ```suggestion
The `--application-id` flag sets the unique identifier for an edge application. It's a required field.

#### origin-key

The `--origin-key` flag sets the unique identifier for an origin. It's a required field.

```

### Optional flags

#### name

The `--name` flag sets the name of the origin being updated.

#### addresses

The `--addresses` flag passes a list of addresses linked to the origin. Multiple addresses should be separated by commas.

#### host-header

The `--host-header` flag specifies the hostname of the server being accessed.

#### origin-type

> The `--origin-type` flag identifies the source of a record, i.e., `single_origin`.

Reviewer: ```suggestion
The `--origin-type` flag identifies the source of a record. Example: `single_origin`.
```

#### origin-protocol-policy

The `--origin-protocol-policy` flag tells the protocol policy used in the origin.

#### origin-path

The `--origin-path` flag defines the path to be appended to the URI when forwarding the request to the origin. Leave it blank to use only the URI.

#### hmac-authentication

The `--hmac-authentication` flag specifies whether Hmac Authentication is used or not.

#### file

The `--file` flag specifies the path to a JSON file containing the attributes of the origin that will be updated. You can use `-` for reading from stdin.

#### hmac-access-key

The `--hmac-access-key` flag informs the Hmac Access Key.

#### hmac-region-name

The `--hmac-region-name` flag informs the Hmac region name.

#### hmac-secret-key

The `--hmac-secret-key` flag informs the Hmac Secret Key.

### help

The `-h` or `--help` option displays more information about the `azion update origin` action.

---

## Delete 

### Usage

```bash
$ azion delete origin --application-id <application-id> --origin-key <origin-key>
```

### Required flags

#### application-id

> The `--application-id` flag sets the unique identifier for an edge application. It is a required field.
> 
> #### origin-key
> 
> The `--origin-key` flag sets the unique identifier for an origin. It is a required field.

Reviewer: ```suggestion
The `--application-id` flag sets the unique identifier for an edge application. It's a required field.

#### origin-key

The `--origin-key` flag sets the unique identifier for an origin. It's a required field.
```
